Website (HTML+CSS+JS)

ZoyaMi opened the pull request:
Задание 4 (проверочный браузер - Chrome): 
Cоздать HTML страницу, содержащую следующие элементы (в ряд):
1) дроп-даун Team, содержащий hard-coded список направлений на проекте ("Main Releases", "Mobile", "Web", дефолтное значение - пустое)
2) дроп-даун Members, содержащий hard-coded списки людей (фамилия имя) для каждого направления (список людей хранится в массиве на клиенте и подтягивается сразу после выбора значения или его изменения в дроп-дауне Team)
3) Generate report кнопка, которая составляет сводную таблицу из выбранных ранее значений (в случае, если нет ошибок): - название таблицы (1 строка, объединяющая все столбцы) = название команды - первая колонка: члены команды (если задан, то вывести репорт для 1 член, если пусто - взять всех членов команды из массива на клиенте и отсортировать по алфавиту) - вторая колонка: статус по каждому члену команды (статусы не хранятся в массиве, а генерятся скриптом по принципу: если в фамилии нечетное кол-во букв - статус "Available", если четное - "Not Available")
Задать в CSS следующие стили для элементов (все стили должны быть в одном CSS документе, отдельно от HTML страницы):
1) Страница: фон
2) Дроп-дауны: border, шрифт
3) Кнопки: фон, шрифт
4) Статусы: "Available" - шрифт, "Not Available" - шрифт
5) Ошибки: шрифт красный, жирный, центрирован по горизонтали
Создать следующие JS валидации (все скрипты должны быть в одном JS файле, отдельно от HTML страницы):
1) Если дроп-даун Team пустой, при попытке создать репорт выдавать ошибку "Please select a team", не отрисовывать таблицу
2) Если в дроп-дауне Members выбран человек, который не числится в выбранной команде (проверять по массиву), выдавать ошибку: "Specified team member doesn't exist in the selected team" (продумать, когда возможен такой сценарий), не отрисовывать таблицу
3) Если в дроп-дауне Team или в дроп-дауне Members выбрано несуществующее значение (такого нет в массиве), выдавать ошибку "No data found", не отрисовывать таблицу

ZoyaMi committed:
Website (HTML+CSS+JS)

SergeBondar commented on Website/Style.css:
```diff
@@ -0,0 +1,19 @@
+body {
+background: #A9F5F9;
```
отступы

SergeBondar commented on Website/index.html:
```diff
@@ -0,0 +1,27 @@
+         <link rel="stylesheet" href="Style.css">
+   </head>
+   <body>
+      Team
```
давай используем <label for="">

SergeBondar replied:
я имел ввиду <label for="">

SergeBondar commented on Website/index.html:
```diff
@@ -0,0 +1,27 @@
+   </head>
+   <body>
+      Team
+      <select class="dropD" id="sel" onchange="getTeamMembers()" >
```
давай имя классу и id получше придумаем

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+document.getElementById("sel").value = '';
```
лучше сразу в разметке создать первой опцией Please select... с value="" и атрибутом selected

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+document.getElementById("sel").value = '';
+var Names1 = ['Jack White', 'Bob Dillan', 'Mark Lee', 'Vera Chang'];
```
camelCase

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+function generateMembersList(Names) {
+    document.getElementById("selName").innerHTML = "";
+    Names.forEach(function(e) {
+        document.getElementById("selName").innerHTML += "<option>" + e + "</option>";
```
еще бы хорошо что-то в value проставить

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+
+function wrongTeamMate(member, team) {
+    for (var i = 0; i < team.length; i++) {
+        if (team[i] == member) {
```
строгое

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+function wrongTeamMate(member, team) {
+    for (var i = 0; i < team.length; i++) {
+        if (team[i] == member) {
+            return false
```
; всегда в конце выражения

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+function getTeamMembers() {
+    var tmp = document.getElementById("sel").value;
+    if (tmp == "Main Releases") {
+        generateMembersList(Names1);
```
как насчет switch?

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+    } else if (tmp == "Web") {
+        generateMembersList(Names3);
+        currentNames = Names3;
+    }
```
можно в switch просто логику по выбору источника оставить (currentNames = ...)
а уже потом отсортировать и вызвать функцию generateMembersList

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+
+function generateReport() {
+    if (!document.getElementById("sel").value) {
+        document.getElementById("svTable").innerHTML = "<div class='error'>Please select a team<div>";
```
посмотри как это можно сделать через document.createElement

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+        }
+    } else if (!document.getElementById("selName").value) {
+        currentNames.forEach(function(e) {;
+            if (e.length % 2 == 0) {
```
дублирующуюся логику надо вынести в отдельную функцию

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,72 @@
+    document.getElementById("selName").value = '';
+}
+
+function wrongTeamMate(member, team) {
```
isWrongTeamMate?

ZoyaMi committed:
Добавлены отступы

ZoyaMi committed:
Добавлена дефолтовая опция

Добавлена дефолтовая опция - <option selected value="">Please select...</option>

ZoyaMi committed:
Добавлены правки

1. Switch вместо 
function getTeamMembers() {
    var tmp = document.getElementById("sel").value;
    if (tmp == "Main Releases") {
        generateMembersList(Names1);
2. document.createElement вместо document.getElementById
3. названия классов изменены
4. переменные через camelCase

SergeBondar commented on Website/index.html:
```diff
@@ -0,0 +1,16 @@
+Team
```
в html документе должны быть html, head, body

SergeBondar commented on Website/js.js:
```diff
@@ -0,0 +1,78 @@
+		var my_div = document.getElementById("svTable");
+		document.body.insertBefore(newDiv, my_div);
+		setTimeout(function () {
+			document.getElementById('errM').remove()
```
; в конце каждого выражения

ZoyaMi committed:
Добавлены head, doctype, body etc.
